lib/routes/xiaoheihe/user.js

```diff
@@ -1,40 +1,50 @@
 const got = require('@/utils/got');
 const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
 
 module.exports = async (ctx) => {
     const id = ctx.params.id;
-
     const response = await got({
         method: 'get',
         url: `https://api.xiaoheihe.cn/bbs/web/profile/post/links?userid=${id}&limit=10&offset=0&version=999.0.0`,
     });
-    const user_data = response.data.user;
-    const post_data = response.data.post_links;
-    const user_name = user_data.username;
-
-    const out = await Promise.all(
-        post_data.map(async (post) => {
-            const title = post.description;
-            const date = post.create_at;
-            const itemUrl = `https://xiaoheihe.cn/community/7216/list/${post.linkid}`;
+    const data = response.data;
+    const username = data.user.username;
+    const articleList = data.post_links.map((post) => ({
+        title: post.title,
+        pubDate: parseDate(post.create_at * 1000),
+        link: post.share_url,
+        linkId: post.linkid
+    }));
 
-            const response = await got.get(itemUrl);
-            const $ = cheerio.load(response.data);
-            const description = $('.post-content').html();
+    const items = await Promise.all(
+        articleList.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const url = `https://api.xiaoheihe.cn/bbs/app/api/share/data/?offset=0&limit=3&link_id=${item.linkId}&os_type=web`;
+                const response = await got.get(url);
+                const link = JSON.parse(response.body).link;
```

Comment:
https://github.com/DIYgod/RSSHub/blob/master/lib/utils/got.js#L16

为什么要自己 parse？


```suggestion
                const link = response.data.link;
```